### PR TITLE
array reduce | Adam Park

### PR DESCRIPTION
**Pull request by Adam-SungMin-Park**
Screen Shot 2021-01-14 at 18 43 28

### CODE REVIEW

File: array-reduce/main.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      saving.push(account[i].amount);
+    }
+    if (account[i].type === 'withdrawal') {
+      saving.push(account[i].amount * -1);
+    }
+  } return saving.reduce(summer);
+};
+
+const balance = account.reduce(bank);
+console.log(balance);
```
Comment (scbowler) [inline array-reduce/main.js:41]: You don't need a separate `for` loop here, `reduce` is already looping through the data. However you are pretty close to the correct solution. Here is an example to help nudge you in the right direction:

```JAVASCRIPT
const balance = account.reduce((total, transaction) => {
  // If the transaction is a deposit add the "amount" to the "total"
  
  // if the transaction is a withdrawal subtract the "amount" from the "total"
  
  // return the "total"
}, 0)
```

**Review by scbowler — CHANGES_REQUESTED** [review]
Everything but the "bank" look good! But some changes are needed for the bank, see below comment for more details.

File: array-reduce/main.js (hunk 1 above)
Comment (Adam-SungMin-Park) [reply array-reduce/main.js]: thanks for your help! just updated 👍

**Review by Zhiwutian — APPROVED** [review]
👍

**Pull request merged**